splunk_add_on_ucc_framework/install_python_libraries.py

```diff
@@ -58,30 +71,83 @@ def _subprocess_call(
 def _pip_install(installer: str, command: str, command_desc: str) -> None:
     cmd = f"{installer} -m pip install {command}"
     try:
-        return_code = _subprocess_call(command=cmd, command_desc=command_desc)
+        return_code = _subprocess_run(command=cmd, command_desc=command_desc).returncode
         if return_code != 0:
             raise CouldNotInstallRequirements
     except OSError as e:
         raise CouldNotInstallRequirements from e
 
 
 def _pip_is_lib_installed(
-    installer: str, target: str, libname: str, version: Optional[str] = None
+    installer: str,
+    target: str,
+    libname: str,
+    version: Optional[str] = None,
+    allow_higher_version: bool = False,
 ) -> bool:
+    if not version and allow_higher_version:
+        raise InvalidArguments(
+            "Parameter 'allow_higher_version' can not be set to True if 'version' parameter is not provided"
+        )
+
     lib_installed_cmd = f"{installer} -m pip show --version {libname}"
-    lib_version_match_cmd = f'{lib_installed_cmd} | grep "Version: {version}"'
 
-    cmd = lib_version_match_cmd if version else lib_installed_cmd
+    if version and allow_higher_version:
+        cmd = f'{lib_installed_cmd} | grep "Version"'
+    elif version and not allow_higher_version:
+        cmd = f'{lib_installed_cmd} | grep "Version: {version}"'
+    else:
+        cmd = lib_installed_cmd
 
     try:
         my_env = os.environ.copy()
         my_env["PYTHONPATH"] = target
-        return_code = _subprocess_call(command=cmd, env=my_env)
-        return return_code == 0
+        if allow_higher_version:
+            result = _subprocess_run(command=cmd, env=my_env)
+            if result.returncode != 0:
+                return False
+            result_version = result.stdout.decode("utf-8").split("Version:")[1].strip()
+            return Version(result_version) >= Version(version)
+        else:
+            return_code = _subprocess_run(command=cmd, env=my_env).returncode
+            return return_code == 0
     except OSError as e:
         raise CouldNotInstallRequirements from e
 
 
+def _check_ucc_library_in_requirements_file(path_to_requirements: str) -> bool:
```

Comment:
This was removed in https://github.com/splunk/addonfactory-ucc-generator/pull/1387.

Reply:
Sorry, my bad. I will remove it.